### DISCUSSION TITLE
Wrapping scikit-learn

### DISCUSSION THREAD

**Discussion by sylvaincom** [probabl-ai/skore#864] [General]
Currently (skore v0.4), we have 2 main features for enhancing scikit-learn: cross-validation and train-test-split. The way we do things is to wrap them. See the examples:
- https://skore.probabl.ai/dev/auto_examples/02_diagnose/plot_01_cross_validate.html
- https://skore.probabl.ai/dev/auto_examples/02_diagnose/plot_02_train_test_split.html

@adrinjalali made the following comment:
> I think my main takeaway is that the current way of doing things (implementing same functions as sklearn) is not the way to go

**Comment by adrinjalali** [comment]
As a user, I find it odd that I need to run functions from `skore` to get diagnostics / checks while I write code.

As a workflow, I'd like to run my code and experiments as usual, and then have a tool which looks at my stored data and model in the project, as well as my code which has stored those values, and give me information about what I could do better like a linter such as mypy does.

**Comment by MarieSacksick** [comment]
We are getting away from this first approach we started, so I guess we can close this discussion.
